smcore/factory.cpp

```diff
@@ -22,19 +22,32 @@
 
 #include "factory.h"
 
+#include <map>
+
 namespace smos
 {
     namespace smcore
     {
         //******************************************************************************
         Options *Factory::getOptions(smos::smcore::SMString optionsfile)
         {
-            static Options *smOptions = nullptr;
-            if (nullptr == smOptions)
+            static std::map<smos::smcore::SMString, Options *> programOptions;
+            if (programOptions.find(optionsfile) == programOptions.end())
             {
-                smOptions = new Options(optionsfile);
+                programOptions.insert(std::make_pair(optionsfile, new Options(optionsfile)));
```

Comment:
Here we have a similar problem as in line 47 - memory leak. the Options object will never be destroyed.

Reply:
See the comments for line 47.